zerver/views/messages.py

```diff
@@ -910,9 +910,12 @@ def update_message_backend(request, user_profile,
         if content == "":
             raise JsonableError(_("Content can't be empty"))
         content = truncate_body(content)
-
+        ums = UserMessage.objects.filter(message=message.id)
```

Comment:
We should add a comment here explaining:

"We don't filter for historical=False here, even though users with UserMessage rows with historical=False were not direct recipients of this message, because those users will still see this message in their feed." (or something similar)

(search for historical in `do_update_message_flags` to see what I'm talking about -- that's the one place where we create rows with historical=False).